File: Python Fundamentals/Mid Exam problems/shoot_for_the_win.py
sequence = list(map(int, input().split()))

while True:
    inp = input()
    if inp == 'End':
        print(f"Shot targets: {sequence.count(-1)} -> {' '.join(list(map(str, sequence)))}")
        break

    index = int(inp)
    if 0 <= index <= len(sequence) - 1:
        current_target_value = sequence[index]
        sequence[index] = -1

        for target in range(len(sequence)):
            if sequence[target] != -1:
                if sequence[target] > current_target_value:
                    sequence[target] -= current_target_value
                else:
                    sequence[target] += current_target_value

File: JavaScript Fundamentals/Mid Exams/Mid Exam 3/shootForTheWin.js
function solve(arr) {
    let sequence = arr.shift().split(' ').map(x => Number(x));
    let shotTargets = 0;
    for (let el of arr) {
        if (el == 'End') break;
        let index = Number(el);
        if (index >= 0 && index < sequence.length) {
            let target = sequence[index];
            if (target != -1) {
                sequence[index] = -1;
                shotTargets++;
            }
            for (let i = 0; i < sequence.length; i++) {
                if (sequence[i] > target && sequence[i] != -1) sequence[i] -= target;
                else if (sequence[i] <= target && sequence[i] != -1) sequence[i] += target;
            }
        }
    }
    console.log(`Shot targets: ${shotTargets} -> ${sequence.join(' ')}`);
}
